airflow/www/views.py

```diff
@@ -3339,6 +3339,58 @@ def list(self):
         )
 
 
+class ProviderView(AirflowBaseView):
+    """View to show Airflow Providers"""
+
+    default_view = 'list'
+
+    class_permission_name = permissions.RESOURCE_PROVIDER
+
+    method_permission_name = {
+        'list': 'read',
+    }
+
+    base_permissions = [
+        permissions.ACTION_CAN_READ,
+        permissions.ACTION_CAN_ACCESS_MENU,
+    ]
+
+    @expose('/provider')
+    @auth.has_access(
+        [
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_PROVIDER),
+        ]
+    )
+    def list(self):
+        """List providers."""
+        providers_manager = ProvidersManager()
+
+        providers = []
+        for pi in providers_manager.providers.values():
+            provider_info = pi[1]
+            provider_data = {
+                "package_name": provider_info["package-name"],
+                "description": self._clean_description(provider_info["description"]),
+                "version": pi[0],
+            }
+            providers.append(provider_data)
+
+        title = "Providers"
+        doc_url = get_docs_url("apache-airflow-providers/index.html")
+        return self.render_template(
+            'airflow/providers.html',
+            providers=providers,
+            title=title,
+            doc_url=doc_url,
+        )
+
+    def _clean_description(self, description):
+        cd = re.sub("[`_]", "", description.strip(" \n.").strip("\""))
+        cd = re.sub("<", "<a href=\"", cd)
+        cd = re.sub(">", "\">[site]</a>", cd)
+        return cd
```

Comment:
```suggestion
        import markupsafe  # Should be moved to the top of the file

        def _build_link(matchobj):
            # These were already escaped, so we can use them as-is
            text = matchobj.group(1)
            url = matchobj.group(2)
            return markupsafe.Markup(f'<a href="{url}">{text}</a>')

        safe = markupsafe.escape(description)
        safe = re.sub(r"`(.*) +&lt;(.*)&gt;`__", _build_link, safe)
        safe = re.sub(r"\n", r"<br>", safe)
        return markupsafe.Markup(safe)
```

This will allow you to remove the `safe` filter, plus I think it renders the links more nicely, e.g:
![Screen Shot 2021-08-13 at 10 27 35 AM](https://user-images.githubusercontent.com/66968678/129390631-e7e2f343-aeea-4486-893a-4a61f54ad1c2.png)

![Screen Shot 2021-08-13 at 10 27 45 AM](https://user-images.githubusercontent.com/66968678/129390706-7cc80b4c-2736-4388-a992-993e0d7c7ef3.png)

Reply:
Agree that this looks cleaner and safer than safe in a template 👍

Reply:
Fwiw, this might need a little more effort. 1 providers link doesn't work right with this, and I thought it was the provider itself but after looking closer I dont think thats the case.